service/create/service.go

```diff
@@ -1299,3 +1296,48 @@ func (s *Service) onDelete(obj interface{}) {
 
 	s.logger.Log("info", fmt.Sprintf("cluster '%s' deleted", cluster.Name))
 }
+
+func (s *Service) onUpdate(oldObj, newObj interface{}) {
+	oldCluster := *oldObj.(*awstpr.CustomObject)
+	cluster := *newObj.(*awstpr.CustomObject)
+
+	if err := validateCluster(cluster); err != nil {
+		s.logger.Log("error", "cluster spec is invalid: %s", errgo.Details(err))
+		return
+	}
+
+	oldSize := len(oldCluster.Spec.AWS.Workers)
+	newSize := len(cluster.Spec.AWS.Workers)
+
+	if oldSize == newSize {
+		// We get update events for all sorts of changes. We are currently only
+		// interested in changes to one property, so we ignore all the others.
+		return
+	} else if oldSize > newSize {
+		s.logger.Log("clusterID", cluster.Spec.Cluster.Cluster.ID, "error", "cluster size reduction not supported")
```

Comment:
Nitpick but I think this should be logged as info rather than error. As its just not supported by the operator yet.